### PR TITLE
Add delete collection methods

### PR DESCRIPTION
**Pull request by dtaniwaki**
I implemented delete collection methods so we don't have to call `delete()` one by one. Could you consider merging this change?

REF: https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.11/#delete-collection-pod-v1-core

### CODE REVIEW

File: client/src/main/scala/skuber/api/package.scala [hunk 1]
```diff
@@ -556,6 +556,32 @@ package object client {
        } yield ()
      }
 
+     def deleteCollection[L <: ListResource[_]]()(
```
Comment (doriordan) [inline client/src/main/scala/skuber/api/package.scala]: Although `deleteCollection` is a logical naming choice when considering the Kubernetes REST API reference, I would prefer to use the name `deleteAll` for the method as this is consistent with the Skuber API method naming for the `watchAll` methods.
Ditto with the other new methods.

**Comment by doriordan** [comment]
Thanks for the PR @dtaniwaki 
Looks good - just one review comment re. naming.
(I am also assuming the modified integration test has been run successfully).

**Comment by dtaniwaki** [comment]
@doriordan Thanks for your review! I have updated the code, so could you check it?

**Comment by doriordan** [comment]
LGTM thanks!

**Pull request merged**